apps/server/src/modules/room/api/room-invitation-link.controller.ts

```diff
@@ -0,0 +1,112 @@
+import { UpdateRoomInvitationLinkBodyParams } from './dto/request/update-room-invitation-link.body.params';
+import { RoomInvitationLinkUseLinkResponse } from './dto/response/room-invitation-link-use-link.response';
+import { RoomInvitationLinkResponse } from './dto/response/room-invitation-link.response';
+import { RoomInvitationLinkMapper } from './mapper/room-invitation-link.mapper';
+import { RoomInvitationLinkUc } from './room-invitation-link.uc';
+
+@ApiTags('Room-Invitation-Links')
+@JwtAuthentication()
+@Controller('room-invitation-links')
+export class RoomInvitationLinkController {
+	constructor(private readonly roomInvitationLinkUc: RoomInvitationLinkUc) {}
+
+	@Post()
+	@ApiOperation({ summary: 'Create a new room invitation link' })
+	@ApiResponse({
+		status: HttpStatus.OK,
+		description: 'Returns a new room invitation link',
+		type: RoomInvitationLinkResponse,
+	})
+	@ApiResponse({ status: HttpStatus.BAD_REQUEST, type: ApiValidationError })
+	@ApiResponse({ status: HttpStatus.UNAUTHORIZED, type: UnauthorizedException })
+	@ApiResponse({ status: HttpStatus.FORBIDDEN, type: ForbiddenException })
+	@ApiResponse({ status: '5XX', type: ErrorResponse })
+	public async createRoomInvitationLink(
+		@CurrentUser() currentUser: ICurrentUser,
+		@Body() createRoomParams: CreateRoomInvitationLinkBodyParams
+	): Promise<RoomInvitationLinkResponse> {
+		const roomInvitationLink = await this.roomInvitationLinkUc.createLink(currentUser.userId, createRoomParams);
+
+		const response = RoomInvitationLinkMapper.mapToRoomInvitationLinkResponse(roomInvitationLink);
+
+		return response;
+	}
+
+	@Put(':roomInvitationLinkId')
+	@ApiOperation({ summary: 'Update an existing room invitation link' })
+	@ApiResponse({
+		status: HttpStatus.OK,
+		description: 'Returns the updated room invitation link',
+		type: RoomInvitationLinkResponse,
+	})
+	@ApiResponse({ status: HttpStatus.BAD_REQUEST, type: ApiValidationError })
+	@ApiResponse({ status: HttpStatus.UNAUTHORIZED, type: UnauthorizedException })
+	@ApiResponse({ status: HttpStatus.FORBIDDEN, type: ForbiddenException })
+	@ApiResponse({ status: HttpStatus.NOT_FOUND, type: NotFoundException })
+	@ApiResponse({ status: '5XX', type: ErrorResponse })
+	public async updateLink(
+		@CurrentUser() currentUser: ICurrentUser,
+		@Param() urlParams: RoomInvitationLinkUrlParams,
+		@Body() updateRoomParams: UpdateRoomInvitationLinkBodyParams
+	): Promise<RoomInvitationLinkResponse> {
+		const roomInvitationLink = await this.roomInvitationLinkUc.updateLink(currentUser.userId, {
+			id: urlParams.roomInvitationLinkId,
+			...updateRoomParams,
+		});
+
+		const response = RoomInvitationLinkMapper.mapToRoomInvitationLinkResponse(roomInvitationLink);
+
+		return response;
+	}
+
+	@Delete(':roomInvitationLinkId')
+	@ApiOperation({ summary: 'Delete a room invitation link' })
+	@ApiResponse({ status: HttpStatus.NO_CONTENT, description: 'Deletion successful' })
```

Comment:
just a thought/question: why didn't we use `HttpStatus.OK` here?

Reply:
because 204 is correct here. In general all the basic http status code 200, 300, 400, 500 are just roughly saying what is the result... and then there are more specific codes like 204: ok, but no content.